packages/design-system/src/components/Modal/Modal.tsx

```diff
@@ -123,11 +133,19 @@ function Modal(props: ModalPropsType): ReactElement {
 										</span>
 
 										{secondaryAction && (
-											<ButtonSecondary {...secondaryAction} data-testid="modal.buttons.secondary" />
+											<ButtonSecondary
+												{...secondaryAction}
+												data-testid="modal.buttons.secondary"
+												data-feature="modal.buttons.secondary"
+											/>
 										)}
 
 										{primaryAction && (
-											<PrimaryAction {...primaryAction} data-testid="modal.buttons.primary" />
+											<PrimaryAction
+												{...primaryAction}
+												data-testid="modal.buttons.primary"
+												data-feature="modal.buttons.primary"
```

Comment:
Can't we pass the data-feature we want to with the spread of primaryAction driven by the host ?

Reply:
so why not add it to types ? it make sens, we are adding data attribute on user actions

Reply:
Created local type for data attributes and added it to ButtonPrimitive props.